src/execution_plan/execution_plan.c

```diff
@@ -270,19 +271,36 @@ ExecutionPlan* NewExecutionPlan(RedisModuleCtx *ctx, Graph *g,
                 size_t expCount = 0;
                 AlgebraicExpression **exps = AlgebraicExpression_From_Query(ast, pattern, q, &expCount);
                 
-                TRAVERSE_ORDER order = determineTraverseOrder(q, execution_plan->filter_tree, exps, expCount);
+                TRAVERSE_ORDER order = determineTraverseOrder(q, filter_tree, exps, expCount);
```

Comment:
In the event that neither `src` nor `dest` are filtered, I wonder if `determineTraverseOrder` should check to see if one has a label while the other doesn't (we'd of course prefer to make a `LabelScan` over an `AllNodeScan`). Thoughts?